analytic_tag_dimension_enhanced/models/account_analytic_dimension.py

```diff
@@ -56,7 +56,12 @@ def create_analytic_tags(self):
         self.analytic_tag_ids.filtered(
             lambda l: not l.resource_ref or l.resource_ref._name != model
         ).unlink()
-        tag_res_ids = [x.resource_ref.id for x in self.analytic_tag_ids]
+        tag_res_ids = []
+        for tag in self.analytic_tag_ids:
+            tag_res_ids.append(tag.resource_ref.id)
+            # Update name analytic tag from Ref model.
+            if tag.display_name != tag.resource_ref.display_name:
+                tag.name = tag.resource_ref.display_name
```

Comment:
```suggestion
        tag_res_ids = [x.resource_ref.id for x in self.analytic_tag_ids]
        for tag in self.analytic_tag_ids.filtered(lambda l.display_name != l.resource_ref.display_name)
            tag.name = tag.resource_ref.display_name
```

Reply:
I think it is 2 different things we want to do. By keep it separated, may be more readable. my2cent